src/worker/taskenv.rs

```diff
@@ -77,18 +78,59 @@ impl TaskEnv {
         }
     }
 
+    fn start_inner_task(
+        &mut self,
+        state: &WorkerState,
+        task_ref: TaskRef,
+        end_receiver: oneshot::Receiver<()>,
+    ) {
+        let task_fut = (*state.inner_task_launcher)(&task_ref);
+        let state_ref = state.self_ref();
+
+        tokio::task::spawn_local(async move {
```

Comment:
This seems.. scary :) Maybe we could store `LocalSet` inside the worker and keep track of the spawned futures? It's probably overkill, since the concurrent number of tasks shouldn't be high.

Reply:
I do not understand why you consider it scary? As it is now bounded by number of CPUs, there should not be too much tasks. For reasonable large cluster, there will be probably more tasks from incomming/outcomming connections than for tako tasks.

Reply:
Ok. It's just weird seeing these unbounded spawns. But you are right, it shouldn't be a problem.

Reply:
But what is the alternative? Unbounded select_all? It would have the same problem + non trivial logic.

Reply:
Something like
```
self.futs.push(spawn_local());
```
+ some check that there aren't too many of them.

Reply:
Then you have to make sure that self.futs is updated and add some logic around. I could implement some counter mechanism inside spawn as sanity check, but its fail means that resource manager fatally failed, and everything is burning and exploding anyway :)

Reply:
Yeah it would only be a sanity check anyway :)